### PR TITLE
Uni v3

### PR DESCRIPTION
**Pull request by gurjotd23**
Decoded the following Contracts and events for Uniswap V3 on ethereum:

Factory Contract:
- Pair Created

Pair Contract:
- Mint
- Burn
- Flash Loan
- Swap
- Collect Fees

NFT Positions Manager:
- Increase Liquidity
- Decrease Liquidity
- Collect Fees

### CODE REVIEW

**Review by jagnani73 — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "license": "ISC",
     "dependencies": {
         "@covalenthq/client-sdk": "^0.8.7",
+        "config": "^3.3.11",
```
[the remark this answers is not in the document]
Comment (gurjotd23) [reply package.json:18]: IDK this file even changed tbh

File: services/decoder/protocols/uniswap-v3/uniswap-v3.decoders.ts [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
+                decimals: Token0?.[0]?.contract_decimals,
+                ticker_symbol: Token0?.[0].contract_ticker_symbol,
+                ticker_logo: Token0?.[0]?.logo_url,
+                pretty_quote: Token0?.[0]?.prices?.[0]?.pretty_price,
+            },
+            {
+                heading: "Token 1 Information",
+                value: "0",
+                decimals: Token1?.[0]?.contract_decimals,
+                ticker_symbol: Token1?.[0].contract_ticker_symbol,
```
[the remark this answers is not in the document]
Comment (gurjotd23) [reply services/decoder/protocols/uniswap-v3/uniswap-v3.decoders.ts]: fixed

File: services/decoder/protocols/uniswap-v3/uniswap-v3.decoders.ts [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+            {
+                heading: "amount0",
+                value: decoded.amount0.toString(),
+                type: "text",
+            },
+            {
+                heading: "amount1",
+                value: decoded.amount1.toString(),
+                type: "text",
+            },
```
[the remark this answers is not in the document]
Comment (gurjotd23) [reply services/decoder/protocols/uniswap-v3/uniswap-v3.decoders.ts:254]: So in this, the heading of amount, amount0, amount1, sender & owner should be capitalized, but tickUpper and tickLower can remain the same?

File: services/decoder/protocols/uniswap-v2/uniswap-v2.configs.ts [hunk 4]
```diff
@@ -3,11 +3,28 @@ import { type Configs } from "../../decoder.types";
...
+        is_factory: false,
+        protocol_name: "uniswap-v2",
+        chain_name: "eth-mainnet",
+    },
+    {
+        address: "0x0d4a11d5eeaac28ec3f61d100daf4d40471f1852",
+        is_factory: false,
+        protocol_name: "uniswap-v2",
+        chain_name: "eth-mainnet",
+    },
```
[the remark this answers is not in the document]
Comment (gurjotd23) [reply services/decoder/protocols/uniswap-v2/uniswap-v2.configs.ts:21]: anything to do for this in this PR for now?

[the remark this answers is not in the document]
Comment (jagnani73) [reply services/decoder/protocols/uniswap-v2/uniswap-v2.configs.ts:21]: No no, just confirming a blocker

File: services/decoder/protocols/uniswap-v3/uniswap-v3.decoders.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jagnani73) [reply services/decoder/protocols/uniswap-v3/uniswap-v3.decoders.ts]: Why? tickLower can be Tick Lower

**Review by jagnani73 — APPROVED** [review]

**Pull request merged**